defaults/main.yml

```diff
@@ -4,11 +4,27 @@ prometheus_version: 2.6.1
 prometheus_config_dir: /etc/prometheus
 prometheus_db_dir: /var/lib/prometheus
 
-prometheus_web_listen_address: "0.0.0.0:9090"
+prometheus_web_listen_port: 9090
```

Comment:
Variable `prometheus_web_listen_address` cannot be removed as this is a variable defined by prometheus itself. 

I would advise to use that one and do `split(':')[-1]` to get port and `{{ var | split(':')[0:-1] | join(':')` to get address. Just remember that `:` char can be specified multiple times due to how IPv6 works.

Reply:
Didn't knew that possibility. That said, I think it's more obvious to a reader to have variables instead of python function to build up variables (one for port, one for ip address, one for socket). I also didn't want to mess with variables already in place so I kinda merge my ideas in `prometheus_web_listen_address`. if I had to build this from scratch I would probably made the main variable be `prometheus_web_listen_socket` rather than ip..